pkg/nfd-master/nfd-master.go

```diff
@@ -1440,3 +1404,26 @@ func (m *nfdMaster) filterFeatureAnnotations(annotations map[string]string) map[
 	}
 	return outAnnotations
 }
+
+func (m *nfdMaster) getNode(nodeName string) (*corev1.Node, error) {
+	return m.k8sClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+}
+
+func (m *nfdMaster) getNodes() (*corev1.NodeList, error) {
+	return m.k8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+}
+
+func (m *nfdMaster) patchNode(nodeName string, patches []utils.JsonPatch, subresources ...string) error {
+	if len(patches) > 0 {
+		data, err := json.Marshal(patches)
+		if err == nil {
+			_, err = m.k8sClient.CoreV1().Nodes().Patch(context.TODO(), nodeName, types.JSONPatchType, data, metav1.PatchOptions{})
+		}
+		return err
+	}
+	return nil
```

Comment:
Maybe we can do this, wdyt?
```suggestion
	if len(patches) == 0 {
	   return nil
	}
        data, err := json.Marshal(patches)
        if err == nil {
	        _, err = m.k8sClient.CoreV1().Nodes().Patch(context.TODO(), nodeName, types.JSONPatchType, data, metav1.PatchOptions{})
        }
        return err
```

Reply:
Definitely, looks better this way. I was just blindly copying the function... Now changed as suggested